packages/@aws-cdk/toolkit-lib/lib/api/diff/diff-formatter.ts

```diff
@@ -154,7 +154,7 @@ export class DiffFormatter {
     this.ioHelper = props.ioHelper;
     this.oldTemplate = props.templateInfo.oldTemplate;
     this.newTemplate = props.templateInfo.newTemplate;
-    this.stackName = props.templateInfo.newTemplate.stackName;
+    this.stackName = props.templateInfo.newTemplate.displayName ?? props.templateInfo.newTemplate.stackName;
```

Comment:
this is kind of iffy, but i think its safe. diff formatter is using stack name to differentiate _nested stacks_, and i don't think nested stacks can have a display name clash.